### DISCUSSION TITLE
Nested Widgets, Renderables, and other concept errors…

### DISCUSSION THREAD

**Discussion by formosa** [Textualize/textual#542] [Q&A]
1.  The term "Renderable" comes up a lot, but I can not find a meaningful definition beyond the vague, "something that can be rendered to the console".  Is there a listing of object types that are considered "Renderable" because it appears to me that the only Renderables are Widgets..
2.  Nest Widgets.. How can I place a Panel border around 2 button widgets and not lose functionality?  I can't crack this. I'd love an example of two buttons that respond to mouse events, grouped inside of a framing Panel, forming a parent widget. Mouse over the parent object changes the panel framing color and continuing on to mousing over a contained button causes it to highlight.  None of the examples I've seen leave any breadcrumbs to a solution. And, I'm not savvy enough to piece this together on my own..
3.  What is Bubble?
4.  Can Views contain Views?  If so, how?
5.  What's the relationship between a Message and an Event?
6.  Is it possible to forward messages/events captured in one object to any other object of my choosing? Down the stack to a child element? (Related to question 2)
Any info that could even slightly improve my understanding on these or related concepts would be greatly appreciated.

**Comment by geoff-va** [comment]
This may help with the first question:

- [RenderableType](https://github.com/Textualize/rich/blob/master/rich/console.py#L274)
- [ConsoleRenderable](https://github.com/Textualize/rich/blob/64991335a67926d76f0bc3ed182741e930af59cd/rich/console.py#L264)
- [RichCast](https://github.com/Textualize/rich/blob/64991335a67926d76f0bc3ed182741e930af59cd/rich/console.py#L254)

From what I gather, it's anything that Rich can Render, eg via `__rich__` or `__rich_console__` or basic text strings.
